PEP/1-Optimizer.rst

```diff
@@ -654,19 +659,150 @@ behaviour is achieved.
 
 Performance
 -----------
-Initial benchmarking of ``Optimizer`` for the ``optimize`` benchmark suite are
-listed below. There is a slowdown for those methods (``nelder-mead`, ``bfgs``,
-``l-bfgs-b``) using an ``Optimizer``. However, this slowdown must be considered
-against a wider context:
+Initial benchmarking of ``Optimizer`` for the ``optimize`` asv benchmark suite
+are listed below (Python 3.6, current numpy). 3cb29828 is current master, and
+1bbf5b1a is from the proposed PR. Only those combinations that showed
+significant speedup/slowdown are tabulated. There is a general slowdown for
+those methods (``nelder-mead`, ``bfgs``, ``l-bfgs-b``) using an ``Optimizer``
+in 1bbf5b1a. However, this slowdown must be considered against a wider context:
 
 - maintenance/develop/test for the class based optimizers is much simpler (c.f.
   Knuth's "Premature Optimization" quote)
 - the evaluation time for objective functions posed by users are going to range
   over several orders of magnitude. Some will be sub microsecond, others will be
-  multi-minute. Only those at the very fastest end will be affected.
-- if a slowdown from 2 ms to 2.4 ms is significant, then one may argue that
-  other approaches are needed (such as compiled extensions).
-
+  multi-minute. Only those at the fastest end are affected optimizer overhead.
+- if a slowdown on the order of 2 ms to 3 ms is significant, then one may argue
+  that other approaches are needed (such as compiled extensions).
+
+There is some degree of uncertainty in the actual timings. For example,
+'rosenbrock_tight + CG', or 'LJ + COBYLA' have a slowdown factor of ~1.23, yet
+neither of those solvers use an ``Optimizer``. This may indicate that more
+repeats are necessary to obtain more accurate timings.
+
++---+------------+------------+-------+------------------------+----------------+-----------+
+|   | before     | after      | ratio |  problem               | method         | resource  |
+|   | [3cb29828] | [1bbf5b1a] |       |                        |                |           |
++===+============+============+=======+========================+================+===========+
+| + | 544.76μs   | 1.02ms     | 1.87  | sin1d                  | nelder-mead    | mean_time |
```

Comment:
Instead of the mean time, we should use the minimum time. Other operating systems get in the way and can slow down the function, and the function will execute no faster than the minimum time. When comparing performance that's more reasonable.

I would be interested in seeing a graph of this.

Reply:
Also, could you put the timing script in the repo? I'd be curious to see, and I think others would care too.